Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonGun.cs

```diff
@@ -248,23 +254,36 @@ private void StartReloadProcess()
         _harpoonFiringState = EHarpoonFiringState.Reloading;
 
         StartCoroutine(ReloadHarpoon());
-
-        RuntimeSfxManager.APlayOneShotSfx?
-            .Invoke(FmodSfxEvents.Instance.HarpoonReload, gameObject.transform.position);
     }
 
     /// <summary>
     /// The process in which the harpoon is reloaded
     /// </summary>
     private IEnumerator ReloadHarpoon()
     {
-        float reloadTimeRemaining = _reloadTime;
-        while(reloadTimeRemaining > 0)
+        while (true)
         {
-            reloadTimeRemaining -= Time.deltaTime;
-            yield return null;
+            if (_currentReserveAmmo > 0)
+            {
+                PlayerManager.Instance.InvokeOnHarpoonStartReloadEvent();
+                float reloadTimeRemaining = _reloadTime;
+                while (reloadTimeRemaining > 0)
+                {
+                    reloadTimeRemaining -= Time.deltaTime;
+                    yield return null;
+                }
+
+                _currentReserveAmmo--;
```

Comment:
Perhaps the reserve ammo could be decremented after firing rather than after reloading. That just makes more sense to me

Reply:
I had it like this for the sake of displaying the correct information. If the gun has full ammo, then it will appear as 1/2 for example (1 shot in the gun, 2 in reserve). Then, once the gun is fired, it should say 0/2 until it reloads. Once the gun reloads, it will say 1/1.
If we decremented this immediately after firing, it would go from 1/2 to 0/1 to 1/1.

Reply:
Still disagree, but I can't force you.

Reply:
In the ammo system, there are two numbers that need to be accurately updated

The ammo currently loaded, and the ammo currently in reserve.
When the player shoots, the reserve number *stays the same* since it has not yet been reloaded. It is decremented here since the reserve changes *after* the reload